test/extended/openstack/local-bd-etcd.go

```diff
@@ -0,0 +1,162 @@
+)
+
+const minEtcdDiskSizeGiB = 7
+const etcdBlockDeviceName = "etcd"
+
+var _ = g.Describe("[sig-installer][Suite:openshift/openstack] The OpenStack platform", func() {
+	defer g.GinkgoRecover()
+
+	var computeClient *gophercloud.ServiceClient
+	var ctx context.Context
+	var dc dynamic.Interface
+	var controlPlaneFlavor string
+	var clusterCPMS objx.Map
+	var clientSet *kubernetes.Clientset
+
+	g.BeforeEach(func() {
+		ctx = context.Background()
+
+		g.By("preparing a dynamic client")
+		cfg, err := e2e.LoadConfig()
+		o.Expect(err).NotTo(o.HaveOccurred())
+		dc, err = dynamic.NewForConfig(cfg)
+		o.Expect(err).NotTo(o.HaveOccurred())
+		clientSet, err = e2e.LoadClientset()
+		o.Expect(err).NotTo(o.HaveOccurred())
+
+		computeClient, err = client(serviceCompute)
+		o.Expect(err).NotTo(o.HaveOccurred())
+
+		skipUnlessMachineAPIOperator(ctx, dc, clientSet.CoreV1().Namespaces())
+		skipUnlessEtcdAdditionalBlockDevice(ctx, dc)
+
+	})
+
+	g.It("checks resources when etcd is on an ephemeral block device", func() {
+
+		g.By("checking that the control plane flavor has enough ephemeral storage")
+		{
+			controlPlaneFlavor = clusterCPMS.Get("spec.template.spec.providerSpec.value.flavor").String()
+			o.Expect(controlPlaneFlavor).NotTo(o.BeEmpty())
+			allPages, err := flavors.ListDetail(computeClient, flavors.ListOpts{}).AllPages()
+			o.Expect(err).NotTo(o.HaveOccurred())
+			flavors, err := flavors.ExtractFlavors(allPages)
+			o.Expect(err).NotTo(o.HaveOccurred())
+			for _, flavor := range flavors {
+				if flavor.Name == controlPlaneFlavor {
+					o.Expect(flavor.Ephemeral).To(o.BeNumerically(">=", minEtcdDiskSizeGiB))
+					break
+				}
+			}
+		}
+
+		g.By("checking that control plane machines have the additional ephemeral disk for etcd")
+		{
+			controlPlaneMachines, err := getControlPlaneMachines(ctx, dc)
+			o.Expect(err).NotTo(o.HaveOccurred())
+			o.Expect(len(controlPlaneMachines)).To(o.Equal(3))
+
+			for _, cpMachine := range controlPlaneMachines {
+				o.Expect(checkEtcdDisk(cpMachine)).To(o.Succeed())
+			}
+		}
+
+	})
+})
+
+// getControlPlaneMachines returns the control plane machines.
+func getControlPlaneMachines(ctx context.Context, dc dynamic.Interface) ([]objx.Map, error) {
+	controlPlaneMachines := make([]objx.Map, 0)
+	mc := dc.Resource(schema.GroupVersionResource{
+		Group:    machineAPIGroup,
+		Version:  "v1",
+		Resource: "machines",
+	}).Namespace(machineAPINamespace)
+	obj, err := mc.List(ctx, metav1.ListOptions{})
+	if err != nil {
+		return nil, err
+	}
+	for _, m := range objects(objx.Map(obj.UnstructuredContent()).Get("items")) {
+		if m.Get("machine.openshift.io/cluster-api-machine-role").String() == "master" {
+			controlPlaneMachines = append(controlPlaneMachines, m)
+		}
+	}
+	return controlPlaneMachines, nil
+}
+
+// checkEtcdDisk checks that the machine has an additional block device named etcd with the correct size and storage type.
+func checkEtcdDisk(machine objx.Map) error {
```

Comment:
If this function is only used in this test, consider moving it in the context of the test itself:

```Go
var _ = g.Describe("[sig-installer][Suite:openshift/openstack] The OpenStack platform", func() {
	defer g.GinkgoRecover()
	checkEtcdDisk := func(machine obj.Map) error { 
```

Reply:
FYI I moved `getControlPlaneMachines` too.